pkg/apis/core/validation/validation.go

```diff
@@ -5503,6 +5503,17 @@ func ValidatePodResize(newPod, oldPod *core.Pod, opts PodValidationOptions) fiel
 		allErrs = append(allErrs, field.Invalid(specPath, newPod.Status.QOSClass, "Pod QOS Class may not change as a result of resizing"))
 	}
 
+	// Do not allow removing resource requests/limits on resize.
+	// TODO: Check for sidecar containers too.
```

Comment:
When will this TODO be done?  Why do we need to merge this without it?

Reply:
sidecar support is being added in https://github.com/kubernetes/kubernetes/pull/128367

Reply:
This PR is racing with https://github.com/kubernetes/kubernetes/pull/128367, but this will probably merge first.

Alternatively, we could just add the sidecar check here, since it's just a no-op without https://github.com/kubernetes/kubernetes/pull/128367

Reply:
Added sidecar support